.buildkite/scripts/win_unit_tests.ps1

```diff
@@ -142,10 +154,19 @@ $env:MAGEFILE_CACHE = $magefile
 
 New-Item -ItemType Directory -Force -Path "build"
 
-if ($env:BUILDKITE_PIPELINE_SLUG -eq "beats-xpack-libbeat") {
-    mage -w reader/etw build goUnitTest
-} else {
-    mage build unitTest
+if ($testType -eq "unittest") {
+    if ($env:BUILDKITE_PIPELINE_SLUG -eq "beats-xpack-libbeat") {
+        mage -w reader/etw build goUnitTest
+    } else {
+        mage build unitTest
+    }
+}
+elseif ($testType -eq "systemtest") {
```

Comment:
The name of the file `.buildkite/scripts/win_unit_tests.ps1` is specific about `unit_tests` while it also runs ` mage systemTest`.

What do you think about this proposal that can avoid maintaining the `.buildkite/scripts/win_unit_tests.ps1` but the pipeline itself?


rename the file for `buildkite/scripts/win_mage.ps1`, it's more generic and could accept different mage goals.

`testType` to be renamed for `mageGoals` and then simplify the script with the below snippet:

```
if ($mageGoals -eq "systemTest") {
    google_cloud_auth
}

mage $mageGoals
```

Change the pipeline stages with:

```yaml
# .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh
      - label: ":windows: Windows Unit Tests - {{matrix.image}}"
        command: ".buildkite/scripts/win_mage.ps1 build unitTest"
     ...
      ##  TODO: uncomment when the issue https://github.com/elastic/beats/issues/38142 is solved
      # - label: ":windows: Windows 10 System Tests"
      #   key: "extended-win-10-system-tests"
      #.  command: ".buildkite/scripts/win_mage.ps1 systemTest"

```

```yaml
# .buildkite/scripts/generate_xpack_ libbeat_pipeline.sh
      - label: ":windows: Windows 11 Unit Tests"
        command: ".buildkite/scripts/win_mage.ps1 -w reader/etw build goUnitTest"

```

Reply:
I agree with you that it could be done as you suggest. There is only one remark. We gonna decline to use Windows scripts soon (I hope in 1 week). If it's possible to avoid additional changes right now, let's do that. @v1v WDYT?

Reply:
Do you mean the file `.buildkite/scripts/win_unit_tests.ps1` will not be used anymore soon? I still think the name of the file is not correct for the nature of what it does, so `.buildkite/scripts/win_tests.ps1` could be more meaningful.

Reply:
Correct.
We are going to use our own BK Windows images and @dliappis is working on it.
When it's ready we are going to use command steps with 1-3 commands without using PowerShell scripts.

Reply:
Renaming this script requires a lot of changes in the already prepared dynamic pipelines.

Reply:
I'll then wait for @dliappis 's feedback; if he has no concerns, I'm happy with this.

Reply:
Thanks @v1v for the great feedback (as usual!) The custom Windows images are now ready, so I am good with letting this as is and get rid of many custom scripts and snippets, including the one discussed here, ASAP.